test/utils/noderesourcetopologies/noderesourcetopologies.go

```diff
@@ -0,0 +1,228 @@
+	var updatedNrtList nrtv1alpha1.NodeResourceTopologyList
+	err := wait.Poll(1*time.Second, timeout, func() (bool, error) {
+		err := cli.List(context.TODO(), &updatedNrtList)
+		if err != nil {
+			klog.Errorf("cannot get the NRT List: %v", err)
+			return false, err
+		}
+		klog.Infof("NRT List current ResourceVersion %s reference %s", updatedNrtList.ListMeta.ResourceVersion, ref.ListMeta.ResourceVersion)
+		return (updatedNrtList.ListMeta.ResourceVersion != ref.ListMeta.ResourceVersion), nil
+	})
+	return updatedNrtList, err
+}
+
+func CheckEqualAvailableResources(nrtInitial, nrtUpdated nrtv1alpha1.NodeResourceTopology) (bool, error) {
+	for idx := 0; idx < len(nrtInitial.Zones); idx++ {
+		zoneInitial := &nrtInitial.Zones[idx] // shortcut
+		zoneUpdated, err := findZoneByName(nrtUpdated, zoneInitial.Name)
+		if err != nil {
+			klog.Errorf("missing updated zone %q: %v", zoneInitial.Name, err)
+			return false, err
+		}
+		ok, what, err := checkEqualResourcesInfo(nrtInitial.Name, zoneInitial.Name, zoneInitial.Resources, zoneUpdated.Resources)
+		if err != nil {
+			klog.Errorf("error checking zone %q: %v", zoneInitial.Name, err)
+			return false, err
+		}
+		if !ok {
+			klog.Infof("node %q zone %q resource %q is different", nrtInitial.Name, zoneInitial.Name, what)
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+func CheckZoneConsumedResourcesAtLeast(nrtInitial, nrtUpdated nrtv1alpha1.NodeResourceTopology, required corev1.ResourceList) (string, error) {
+	for idx := 0; idx < len(nrtInitial.Zones); idx++ {
+		zoneInitial := &nrtInitial.Zones[idx] // shortcut
+		zoneUpdated, err := findZoneByName(nrtUpdated, zoneInitial.Name)
+		if err != nil {
+			klog.Errorf("missing updated zone %q: %v", zoneInitial.Name, err)
+			return "", err
+		}
+		ok, err := checkConsumedResourcesAtLeast(zoneInitial.Resources, zoneUpdated.Resources, required)
+		if err != nil {
+			klog.Errorf("error checking zone %q: %v", zoneInitial.Name, err)
+			return "", err
+		}
+		if ok {
+			klog.Infof("match for zone %q", zoneInitial.Name)
+			return zoneInitial.Name, nil
+		}
+	}
+	return "", nil
+}
+
+func checkEqualResourcesInfo(nodeName, zoneName string, resourcesInitial, resourcesUpdated []nrtv1alpha1.ResourceInfo) (bool, string, error) {
+	for _, res := range resourcesInitial {
+		initialQty := res.Available
+		updatedQty, ok := findResourceAvailableByName(resourcesUpdated, res.Name)
+		if !ok {
+			return false, res.Name, fmt.Errorf("resource %q not found in the updated set", res.Name)
+		}
+		if initialQty.Cmp(updatedQty) != 0 {
+			klog.Infof("node %q zone %q resource %q initial=%v updated=%v", nodeName, zoneName, res.Name, initialQty, updatedQty)
+			return false, res.Name, nil
+		}
+	}
+	return true, "", nil
+}
+
+func checkConsumedResourcesAtLeast(resourcesInitial, resourcesUpdated []nrtv1alpha1.ResourceInfo, required corev1.ResourceList) (bool, error) {
+	for resName, resQty := range required {
```

Comment:
Is it "AtLeast" because we are iterating over the required resources?

Reply:
AtLeast because the check isn't precise. We should tolerate some fluctuations in the resource accounting, at least initially, and only if it is safe move to exact equality check. I'm trying to strike a balance between test precision and reliability.

Reply:
While resolving the test conversation I think it could be a good idea to split this utilities in another PR and merge them to use it on the rest of the workload testing. 
@fromanirh what do you think?

Reply:
done: https://github.com/openshift-kni/numaresources-operator/pull/114